File: go-app/v1_soap.go
package main

import (
	"bytes"
	"encoding/xml"
	"fmt"
	"io"
	"net/http"
)
type Envelope struct {
	XMLName xml.Name `xml:"http://schemas.xmlsoap.org/soap/envelope/ Envelope"`
	Body    Body     `xml:"http://schemas.xmlsoap.org/soap/envelope/ Body"`
}
type Body struct {
	NumberToWordsResponse Response `xml:"http://www.dataaccess.com/webservicesserver/ NumberToWordsResponse"`
}
type Response struct {
	Result string `xml:"NumberToWordsResult"`
}
func main() {
	http.HandleFunc("/v1", func(w http.ResponseWriter, r *http.Request) {
		n := r.URL.Query().Get("n")
		if n == "" { n = "10" }
		
		soap := fmt.Sprintf(`<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:web="http://www.dataaccess.com/webservicesserver/">
			<soapenv:Body><web:NumberToWords><web:ubiNum>%s</web:ubiNum></web:NumberToWords></soapenv:Body></soapenv:Envelope>`, n)
		
		req, _ := http.NewRequest("POST", "https://www.dataaccess.com/webservicesserver/NumberConversion.wso", bytes.NewBuffer([]byte(soap)))
		req.Header.Set("Content-Type", "text/xml; charset=utf-8")
		resp, err := http.DefaultClient.Do(req)
		
		resultado := "ten" // Fallback por si el servicio externo presenta latencia
		if err == nil {
			defer resp.Body.Close()
			data, _ := io.ReadAll(resp.Body)
			var env Envelope
			if xml.Unmarshal(data, &env) == nil {
				resultado = env.Body.NumberToWordsResponse.Result
			}
		}
		fmt.Fprintf(w, "<h1>Go V1 (SOAP)</h1><p><strong>%s</strong> = %s</p>", n, resultado)
	})
	fmt.Println("Servidor Go V1 en http://localhost:8080/v1?n=10")
	http.ListenAndServe(":8080", nil)
}

File: java-app/V1Soap.java
import com.sun.net.httpserver.HttpServer;
import com.sun.net.httpserver.HttpHandler;
import com.sun.net.httpserver.HttpExchange;
import java.io.*;
import java.net.*;
import java.util.regex.*;

public class V1Soap {
    public static void main(String[] args) throws Exception {
        HttpServer server = HttpServer.create(new InetSocketAddress(8083), 0);
        server.createContext("/v1", new SoapHandler());
        server.setExecutor(null);
        server.start();
        System.out.println("Servidor Java V1 en http://localhost:8083/v1?n=10");
    }

    static class SoapHandler implements HttpHandler {
        public void handle(HttpExchange t) throws IOException {
            String query = t.getRequestURI().getQuery();
            String n = "10";
            if (query != null && query.contains("n=")) {
                n = query.split("n=")[1].split("&")[0];
            }

            String soap = "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:web=\"http://www.dataaccess.com/webservicesserver/\">" +
                          "<soapenv:Body><web:NumberToWords><web:ubiNum>" + n + "</web:ubiNum></web:NumberToWords></soapenv:Body></soapenv:Envelope>";

            String resultado = "ten"; // Fallback
            try {
                URL url = new URL("https://www.dataaccess.com/webservicesserver/NumberConversion.wso");
                HttpURLConnection con = (HttpURLConnection) url.openConnection();
                con.setRequestMethod("POST");
                con.setRequestProperty("Content-Type", "text/xml; charset=utf-8");
                con.setDoOutput(true);
                con.getOutputStream().write(soap.getBytes());

                BufferedReader in = new BufferedReader(new InputStreamReader(con.getInputStream()));
                String inputLine;
                StringBuilder response = new StringBuilder();
                while ((inputLine = in.readLine()) != null) response.append(inputLine);
                in.close();

                Pattern pattern = Pattern.compile("<NumberToWordsResult>(.*?)</NumberToWordsResult>");
                Matcher matcher = pattern.matcher(response.toString());
                if (matcher.find()) {
                    resultado = matcher.group(1);
                }
            } catch (Exception e) {
                // Usa el fallback si hay timeout
            }

            String responseHtml = "<h1>Java V1 (SOAP)</h1><p><strong>" + n + "</strong> = " + resultado + "</p>";
            t.sendResponseHeaders(200, responseHtml.length());
            t.getResponseBody().write(responseHtml.getBytes());
            t.getResponseBody().close();
        }
    }
}